Style/30 storybook design system

Juliafalk opened the pull request:
**Created design system for DPC Next in Storybook and common components.** 

The design system could be found under: _DPC Next/Design System Docs_
The design system includes an overview, usage and examples. 

For the design system, I've created components just for showing the colors, color palettes and spacing (hence, these will not be used in the project). These can be found under _Design System_ in Storybook. I am not sure where these should be stored, currently stored under framework/design-system. We discussed the naming convention of these, but I am nor sure if I got it correctly. 

Common components: 
- Components for different text types, like heading, subtitles, body text and caption. Properties could be changed, currently just suggestions. Could be found under _typography_
- Component for icons. Currently just a placeholder for icons, hence, the icons need to be imported from heroicons each time being used. I am not sure what is the best way of doing so I would like to discuss/have comments.

Sources:
https://levelup.gitconnected.com/documenting-design-systems-with-storybook-docs-e54b2fa5203
https://storybook.js.org/docs/react/writing-docs/introduction
https://medium.com/storybookjs/rich-docs-with-storybook-mdx-61bc145ae7bc

Questions/appreciated feedback: 
- Where should components used for constructing the design system be stored?
- What should be the naming convention for these components?
- How should the icon component be constructed? Should the icons be imported to this component instead? Should I construct a component for each icon we are using instead?

bjornallvin reviewed (approved):
Awesome job Julia. I made some small changes:
- Aggregated all different text components into a single component called `Text` that takes a type prop for the different styles of texts. Easier to remember and you get intellisense support because its stricter typed. Also scales easier and is easier to maintain.
- Moved 'design-system' folder out from the 'framework' folder. The framework folder should only contain components used in the framework.
- Renamed 'Icons' component to `IconWrapper`.
- Updated the 'Design System' document to reflect theese changes

The color palettes still confuse me a bit. I feel there are to many colors to choose from for thius demo application  :-) But lets discuss this further later.

Juliafalk commented:
Thanks and thanks for the feedback. Regarding the color palette, we could discuss this more further, but I thought this was what we agreed on. Maybe I've misunderstood something.

Thanks for changing those things!

Juliafalk commented:
I made some changes to the design system in Storybook according to what we discussed earlier. 

- Removed the Color Palette.
- Updated Tailwind.config.js with theme colors.
- Renamed the colors in 'primary colors' and 'secondary colors'. The name is defined in tailwind.config.js with the tailwindCSS color name commented on the side (okey to remove, but thought it would be easier if we want to change these colors later).  
- Adjusted the colors for icons and text according to the names in tailwind.config.js.
